components/style/values/mod.rs

```diff
@@ -114,6 +114,29 @@ impl ToCss for CustomIdent {
     }
 }
 
+/// Nonstandard (https://developer.mozilla.org/en-US/docs/Web/CSS/-moz-context-properties)
+#[derive(Clone, Debug, MallocSizeOf, PartialEq, ToComputedValue, ToCss)]
+pub struct MozContextProperties(pub CustomIdent);
+
+impl MozContextProperties {
+    /// Parse an already-tokenizer identifier
+    pub fn from_ident<'i>(location: SourceLocation, ident: &CowRcStr<'i>, excluding: &[&str])
+                          -> Result<Self, ParseError<'i>> {
+        let valid = match_ignore_ascii_case! { ident,
+            "initial" | "inherit" | "unset" | "default" => false,
+            _ => true
+        };
+        if !valid {
+            return Err(location.new_custom_error(SelectorParseErrorKind::UnexpectedIdent(ident.clone())));
+        }
+        if excluding.iter().any(|s| ident.eq_ignore_ascii_case(s)) {
+            Err(location.new_custom_error(StyleParseErrorKind::UnspecifiedError))
+        } else {
+            Ok(MozContextProperties(CustomIdent(Atom::from(ident.as_ref()))))
+        }
+    }
+}
```

Comment:
cc @emilio 
Should we put `MozContextProperties` inside `values/mod.rs` ?
IMO, if this attribute is SVG related, it might be better to put in `values/specified/svg.rs`

Reply:
I was thinking of that too, but putting it inside `values/specified/svg.rs` would require importing `Atom, SelectorParseErrorKind, CustomIdent, SourceLocation, CowRcStr`, and `ParseError`. Should we do it?

Reply:
I'll guess it should be fine if they're needed but you can wait for @emilio 's review.